dart/lib/src/sentry_client.dart

```diff
@@ -280,8 +284,14 @@ class SentryClient {
     // therefore add it to the threads.
     // https://develop.sentry.dev/sdk/event-payloads/stacktrace/
     if (stackTrace != null || _options.attachStacktrace) {
```

Comment:
I know this is not part of this PR but isn't the `attachStacktrace` option basically useless because it will still add teh stacktrace if there is one even though a user might've set `attachStacktrace` to false.

We don't have to address this here but I just find it strange

Reply:
The check was introduced [here](https://github.com/getsentry/sentry-dart/pull/160/files#).

And with the change in [#524](https://github.com/getsentry/sentry-dart/pull/524) the fallback thread was moved from the event to the threads.

Reading the discription for `attachThreads`, I think we should only attach it here if it is set to true. It's a breaking change, as it affects grouping. This feature is also opt-out, so it will not affect as many users.

Seeing that @rxlabz and @ueman were involved, maybe they have some additional insight for us?